geomstats/geometry/hypersphere.py

```diff
@@ -167,45 +169,39 @@ def extrinsic_to_intrinsic_coords(self, point_extrinsic):
 
         return point_intrinsic
 
-    def random_uniform(self, n_samples=1, bound=0.5):
+    def random_uniform(self, n_samples=1):
         """
         Sample in the Hypersphere with the uniform distribution.
 
         Parameters
         ----------
         n_samples : int, optional
-        bound: float, optional
 
         Returns
         -------
-        point : array-like, shape=[n_samples, dimension + 1]
+        samples : array-like, shape=[n_samples, dimension + 1]
         """
-        size = (n_samples, self.dimension)
-
-        if bound is None:
-            spherical_coord = gs.random.rand(*size) * gs.pi
-            spherical_coord[:, -1] *= 2
-
-            point = gs.zeros((n_samples, self.dimension+1))
-            sin_prod = gs.cumprod(gs.sin(spherical_coord), axis=1)
-
-            factor_1 = gs.hstack((gs.ones((n_samples, 1)), sin_prod))
-            factor_2 = gs.hstack(
-                (gs.cos(spherical_coord), gs.ones((n_samples, 1)))
-            )
-
-            point = factor_1 * factor_2
-        else:
-            assert bound <= 0.5
-            point = bound * (2 * gs.random.rand(*size) - 1)
-            point = self.intrinsic_to_extrinsic_coords(point)
-
-        return point
+        size = (n_samples, self.dimension + 1)
+
+        samples = gs.random.normal(size=size)
+        while True:
+            norms = gs.linalg.norm(samples, axis=1)
+            # TODO(nkoep): Should the 'bad sample tolerance' be configurable
+            #              via a function argument? Is TOLERANCE a good
+            #              default?
+            indcs = norms < TOLERANCE
+            num_bad_samples = gs.sum(indcs)
```

Comment:
gs.sum(gs.isclose(norms, 0.))

or RANDOM_TOL

?

Reply:
Jepp, that works :+1: